test/e2e/network/dns_configmap.go

```diff
@@ -217,6 +219,65 @@ func (t *dnsPtrFwdTest) run() {
 		func(actual []string) bool { return len(actual) == 0 },
 		"dnsmasq",
 		moreForeverTestTimeout)
+
+	// Should still be able to lookup public nameserver without upstream nameserver.
+	t.checkDNSRecordFrom(
+		"8.8.8.8.in-addr.arpa",
+		func(actual []string) bool { return len(actual) == 1 && actual[0] == googleDnsHostname+"." },
+		"dnsmasq",
+		moreForeverTestTimeout)
+}
+
+type dnsExternalNameTest struct {
+	dnsTestCommon
+}
+
+func (t *dnsExternalNameTest) run() {
+	t.init()
+
+	t.createUtilPod()
+	defer t.deleteUtilPod()
+
+	t.createDNSServer(map[string]string{
+		googleDnsHostname: "192.0.2.123",
```

Comment:
Instead of using `googleDnsHostname`, can we use something else (`foo.example.com`)? This sounds like hijacking...

Reply:
Changed to foo.example.com so the code looks less like hacking something.